### ISSUE TITLE
Visual Studio 2019 IntelliSense won't finish loading for certain projects

### ISSUE THREAD

**Issue by Gnbrkm41** [dotnet/roslyn#41315]
**Version Used**: 
Visual Studio 2019 Enterprise/Community Preview 16.5 P2. Was able to reproduce on two separate installations where one was a fresh installation.
**Steps to Reproduce**:

1. Clone `dotnet/runtime` repo
2. Locate any of the solution files under `src/libraries/*` folder (While I haven't checked all of them, all of random ones I've opened seemed to suffer from the same problem; to list a few, `System.IO.FileSystem`, `System.Net.Sockets`, `System.Runtime` and `System.Text.Json`)
3. Open the solution via `devenv` - `Open a project or solution` or `devenv SolutionName.sln`

**Expected Behavior**:
IntelliSense should finish loading eventually in the background tasks list

**Actual Behavior**:
It gets stuck loading indefinitely it seems; even after two to three hours of waiting it did not finish loading. It gets stuck at `Loading <n-1>/<n>`, `Loading <n>/<n>` or `Load pending` as shown below:
![image](https://user-images.githubusercontent.com/42944058/73477687-14d93e80-43d8-11ea-8a7d-8bc13892b308.png)
![image](https://user-images.githubusercontent.com/42944058/73477689-1571d500-43d8-11ea-8696-4c15a0dcdbe1.png)
![image](https://user-images.githubusercontent.com/42944058/73477893-68e42300-43d8-11ea-95f8-be0bf766fa9b.png)

It is possible that the projects have something in common and that is causing problems, but it feels rather odd to just silently get stuck here.

Attaching VS logs: 
[Opened through the menu (`devenv /log ...`), when it got stuck at "Loading 2/3"](https://github.com/dotnet/roslyn/files/4135653/menu.txt)

[Opened directly (`devenv src/libraries/System.IO.FileSystem/System.IO.FileSystem.sln /log ...`, when it said "Load pending")](https://github.com/dotnet/roslyn/files/4135654/direct.txt)

(I'm not sure if opening through the menu or opening directly affects on where it gets stuck at. It appears to be somewhat random)

<details><summary>VS version / Installed workloads</summary>

```
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.5.0 Preview 2.0
VisualStudio.16.Preview/16.5.0-pre.2.0+29721.120
Microsoft .NET Framework
Version 4.8.03921

Installed Version: Enterprise

Visual C++ 2019   00435-60000-00000-AA271
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.5.170.20430
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   16.5.170.20430
Azure App Service Tools v3.0.0

C# Tools   3.5.0-beta2-20057-04+f43cfcd2be1b781e0b5a9153776f7e5fecf92b89
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   5.5.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Syntax Visualizer   1.0
An extension for visualizing Roslyn SyntaxTrees.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.20113.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.5.0-beta2-20057-04+f43cfcd2be1b781e0b5a9153776f7e5fecf92b89
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.8.0.0 for F# 4.7   16.5.0-beta.20071.14+c4c09df4d5a68c3e9a51948bab554d11e2c141f2
Microsoft Visual F# Tools 10.8.0.0 for F# 4.7

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake
```
</details>

**Comment by jinujoseph** [comment]
cc @dibarbet can you work with Mohit on this

**Comment by dibarbet** [comment]
Dupe of https://github.com/dotnet/project-system/issues/5834
and https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1052518/
Should be resolved in next 16.5 preview

**Comment by RolandJochems** [comment]
I updated to visual studio 16.5.0 today. And now I get waiting for intellisense to finish, Also publishing projects that use GitVersionTask fail because " The "WriteVersionInfoToBuildLog" task failed unexpectedly." It worked perfectly until I upgraded visual studio.

**Comment by martosource** [comment]
I have also just upgraded to 16.5.0 and I am getting the same waiting for intellisense to finish message as @RolandJochems

**Comment by johncburns1** [comment]
Sam issue as well.  Newest version of VS Preview works.

**Comment by xpeH** [comment]
I have also just upgraded to 16.5.0 and got this issue. After updated to 16.5.1 - same result as well

**Comment by martosource** [comment]
@xpeH I was able to fix it for as I had a .cshtml and .razor file with the same name.

**Comment by johncburns1** [comment]
@martosource interesting. This might be the crux of the issue for me. Intellisense only gets caught on certain Blazor projects for me and my fix was to just unload them from my solution. It sounds like your issues were Blazor related as well. I am going to try your fix and I will report my findings.

**Comment by nigelgandy** [comment]
I upgraded to VS 2019 Pro 16.5.1 and have the same issue ... also cannot clean or build projects (even new console projects) ... uninstalling all of VS and reinstalling

**Comment by dibarbet** [comment]
cc @davkean

**Comment by Goomig** [comment]
Also experiencing this issue +1

**Comment by davkean** [comment]
Folks that are running into this, please Help -> Send Feedback -> Report a Problem making sure to record the instance of Visual Studio that has the problem while it has the problem. This lets us grab logs & dumps that will help us figure out the cause.

**Comment by johnguilbert1970** [comment]
Same issue here.

**Comment by davkean** [comment]
@johnguilbert1970 Can you point me to your Developer Feedback item?

**Comment by pranshum** [comment]
@davkean My report can be found [here](https://developercommunity.visualstudio.com/content/problem/965845/intellisense-wont-finish-loading-for-certain-proje.html)

**Comment by MikaelFerland** [comment]
I tried with 16.5.2, it seems to be stuck in Loading Intellisense - Loading 231/233 XXX.csproj and even if I reload the project both numbers 231/233 are just going higher. If I only unload the project it stuck on another one.

**Comment by davkean** [comment]
Thanks @pranshum.

@MikaelFerland Can you point me to your Developer feedback item?

**Comment by codehunter13** [comment]
same here

**Comment by MikaelFerland** [comment]
@davkean https://developercommunity.visualstudio.com/content/problem/973105/intellisense-wont-finish-loading-for-certain-proje-1.html

**Comment by johnguilbert1970** [comment]
Sorted this. Had to close open document windows and it fixed issue. Not sure if there were duplicate windows. It's still a bug though.
 
 
-----Original Message-----

**Comment by MikaelFerland** [comment]
I found a workaround, I unchecked the feature here :
Environment->Preview Features->Load project faster

Otherwise, the Intellisense never end is loading.

**Comment by pranshum** [comment]
For me, a Blazor App project prevented IntelliSense from loading.

I uninstalled ReSharper - had no luck. 
Deleted all VS caches, reinstalled Visual Studio - no luck again.

Finally, I managed to workaround the issue by delete empty `.g.cs` files from the Blazor App project (auto generated files for razor components) and upgrading the package
`Microsoft.VisualStudio.Azure.Containers.Tools.Targets` to version 1.10.8 in the Blazor App. A quick restart of Visual Studio and things were seemingly back to some form of normality...

Curiously enough, this issue only started happening after upgrading VS to 16.5.1.

**Comment by tomcarter** [comment]
Had the problem with 16.5.1 and resx files.  Upgraded to 16.5.2 which seemed to fix it

**Comment by MuhammadJazabDMT** [comment]
am having same issue and not able to solve it.

**Comment by PylotLight** [comment]
Also having the same issue with my Blazor projects on VS 2019 preview 16.6

**Comment by hrx521** [comment]
Also having the same issue , Blazor webassembly projects , VS 2019 16.6.0 Preview 2.1.
Microsoft.AspNetCore.Components.WebAssembly.Runtime - 3.2.0-preview3.20168.1

**Comment by DaveCousineau** [comment]
Had this issue VS2019 16.5.2. Libraries/desktop projects would hang forever loading project properties window saying waiting for Intellisense to finish loading. ASP.NET projects wouldn't do anything when choosing properties.

Turning off Load Projects Faster fixed the problem as mentioned by MikaelFerland.

**Comment by MagicPelikan** [comment]
Turning off Load Projects Faster as mentioned by MikaelFerland worked for me too. Thank you!

**Comment by pranshum** [comment]
Seems like they have rolled out a [fix for it](https://developercommunity.visualstudio.com/content/problem/965845/intellisense-wont-finish-loading-for-certain-proje.html)

**Comment by Jehoel** [comment]
I'm having this issue when I go Main Menu > Test > Analyze Code Coverage for All Tests. I also get prompted to close the `.coverage` file immediately after it opens, which is weird.

**Comment by mbpakalin** [comment]
Problem still persists.

**Comment by DariuszKalita** [comment]
No IntelliSense support neither in xaml nor in code. Turning off Load Projects Faster doesn't fix it. VS 16.6.0, Xamarin project.

**Comment by DigitalZilla** [comment]
Neither disabling _Load Projects Faster_, nor updating to 16.6 (with _Load Projects Faster_ enabled or disabled) fixed the issue for me. 

As you can see, I have a very simple class - I'm using [ReactiveExtensions ](https://reactiveui.net/docs/getting-started/compelling-example)to set up some observables. lambda _x_ is _this_, which is the **AppViewModel**. The compiler and mouse-over correctly finds my class property (TermResult), but it won't show up in the intellisense auto-complete.

In case it's related, there's a massive amount of UI lag when trying to use the arrow keys to navigate the intellisense autocomplete drop down. Sometimes, pressing up or down will take over a second to actually move the selection, despite the rest of my typing experience being perfectly responsive.

![image](https://user-images.githubusercontent.com/18022018/82851372-74da8000-9eb4-11ea-9ee6-21c27c5410e7.png)

**Comment by AntvissMedia** [comment]
Still occuring in 16.6.2. Turning off Load Projects Faste as mentioned by MikaelFerland still works as a workaround.

**Comment by 3esam** [comment]
Still occurring in 16.6.3
MikaelFerland workaround works.
Also the magical "run as administrator" worked for me 😈

**Comment by alohaninja** [comment]
Was occurring for me in 16.6.3 - seems to be fixed now in 16.7.2

**Comment by Reelix** [comment]
Still broken in 16.8.0 Preview 3 - The workaround (Disabling "Load projects faster") solved it.

**Comment by adrianvmsft** [comment]
Could you please file a feedback item and attach a heap dump? If possible, please provide a title as specific as possible, to be able to distinguish from other similar issues.

These instructions explain how to do that:
https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2019

Thank you,
Adrian

**Comment by qoobes** [comment]
This still happens in 2021. 

I fixed my own issue by disabling **Faster Loading** in **Experimental Features**

**Comment by fruitbatinshades** [comment]
In my case this started AFTER I installed the VS2019 professional preview but it affected the main VS2019 Professional. Turning off Load Projects Faster in preview fixed it in the normal one. Wonder if there is a config thing going?

**Comment by vilayatk** [comment]
It worked for me finally. I tried a couple of things, unchecking the "Load projects faster" option from Environment - > Preview features did not work for me. There was a hidden folder ".vs" in my solution, deleting it did the job!

**Comment by sgf** [comment]
This is a vicious bug that has existed since VS2017, but the strange thing is that it is not always triggered, and once the bug is triggered when a certain solution or project is compiled, it will always be stuck. And this Problems like the VS official feedback have never been solved in a more fundamental way because they cannot be reproduced at all.

**Comment by sgf** [comment]
After I started VS2022 with administrator rights, the problem seemed to disappear.

**Comment by CyrusNajmabadi** [comment]
It's worth noting that this is not roslyn. It's the project system. There's not much we can do about this over here.

**Comment by ArmorBearer** [comment]
This issue should be reopened, as it still occurs - for me, in VS 2022.

I found that it happens when trying to build my project.  I have to cancel the build when it happens.

Fortunately, it's not (currently) happening when I Rebuild All, but I am not able to debug my code, because it tries (unnecessarily) to build the project again before starting the App.  And that's when it gets completely stuck.

I tried turning off Load Projects Faster in preview settings, but that didn't help.

So I thought perhaps I needed to close VS in order for the settings to take effect, but VS just hung trying to unload the first project.  I had to use Task Manager to kill the VS process.

Here's what I saw in Task Manager:
![Image](https://github.com/user-attachments/assets/0b00687b-626e-4fb3-b3c2-d38a8079fbca)

Notice that in addition to the VS process, there is a RoslynCodeAnalysis process running, with 795.5MB of memory usage. *Why do I need my code analyzed when I asked VS to shut down?!?!?*

(I also think it's odd that MS Edge has 9 subprocesses running, using 315.6MB when I don't even have it open! And if something is using Edge's services in the background, I would expect it to be listed under Background processes, not Apps!)
